pkg/utils/remediation_diff.go

```diff
@@ -1,44 +1,287 @@
+		pr.Annotations = annotateErrorStatus("Check sources differ in more than status\n" + diffMsg)
+		pr.Remediation = nil
+	} else {
+		pr.CheckResult.Status = compv1alpha1.CheckResultInconsistent
+		pr.Annotations, createRemediations = annotateInconsistentStatuses(inconsistent)
+		if !createRemediations {
+			pr.Remediation = nil
 		}
+	}
+
+	pr.Labels = make(map[string]string)
+	pr.Labels[compv1alpha1.ComplianceCheckInconsistentLabel] = ""
+
+	return &pr
+}
 
-		ok = diffRemediations(oldList[i].Remediation, newList[i].Remediation)
+func differsExceptStatus(inconsistent []*ParseResultContextItem) (bool, string) {
+	if len(inconsistent) < 2 {
+		return false, ""
+	}
+	base := inconsistent[0]
+
+	for _, item := range inconsistent[1:] {
+		ok := cmp.Equal(base, item,
+			cmpopts.IgnoreTypes(compv1alpha1.ComplianceCheckResult{}),
+			cmpopts.IgnoreUnexported(ParseResultContextItem{}))
 		if !ok {
-			return false
+			diff := cmp.Diff(base, item,
+				cmpopts.IgnoreTypes(compv1alpha1.ComplianceCheckResult{}),
+				cmpopts.IgnoreUnexported(ParseResultContextItem{}))
+			return true, diff
+		}
+	}
+
+	return false, ""
+}
+
+func annotateErrorStatus(msg string) map[string]string {
+	annotations := make(map[string]string)
+	annotations[compv1alpha1.ComplianceCheckResultErrorAnnotation] = msg
+	return annotations
+}
+
+func annotateInconsistentStatuses(inconsistent []*ParseResultContextItem) (map[string]string, bool) {
+	mostCommonState, hasCommonState := mostCommonState(inconsistent)
+	createRemediation := true
+
+	annotations := make(map[string]string)
+	for _, check := range inconsistent {
+		// We'll only create remediations for inconsistent result that contain pass,fail or info
+		// as they still can be remediatied
+		switch check.CheckResult.Status {
+		case compv1alpha1.CheckResultFail, compv1alpha1.CheckResultPass, compv1alpha1.CheckResultInfo:
+			break
+		default:
+			createRemediation = false
+		}
+
+		if hasCommonState && check.CheckResult.Status == mostCommonState {
+			continue
+		}
+
+		for _, src := range check.sources {
+			curVal, ok := annotations[compv1alpha1.ComplianceCheckResultInconsistentSourceAnnotation]
+			if !ok {
+				annotations[compv1alpha1.ComplianceCheckResultInconsistentSourceAnnotation] = src + ":" + string(check.CheckResult.Status)
+			} else {
+				annotations[compv1alpha1.ComplianceCheckResultInconsistentSourceAnnotation] = curVal + "," + src + ":" + string(check.CheckResult.Status)
+			}
+		}
+	}
+
+	if hasCommonState {
+		annotations[compv1alpha1.ComplianceCheckResultMostCommonAnnotation] = string(mostCommonState)
+	}
+
+	return annotations, createRemediation
+}
+
+func mostCommonState(inconsistent []*ParseResultContextItem) (compv1alpha1.ComplianceCheckStatus, bool) {
+	statusCounter := make(map[compv1alpha1.ComplianceCheckStatus]int)
+	for _, check := range inconsistent {
+		statusCounter[check.CheckResult.Status] = statusCounter[check.CheckResult.Status] + len(check.sources)
```

Comment:
Here we were always increasing by one which doesn't work if we first get several passing results and then one failing.